operator/cmd/root.go

```diff
@@ -650,6 +650,15 @@ func (legacy *legacyOnLeader) onStart(_ cell.HookContext) error {
 	ciliumNodeSynchronizer := newCiliumNodeSynchronizer(legacy.clientset, nodeManager, withKVStore)
 
 	if legacy.clientset.IsEnabled() {
+		// ciliumNodeSynchronizer uses operatorWatchers.PodStore for IPAM surge
+		// allocation. Initializing PodStore from Pod resource is temporary until
+		// ciliumNodeSynchronizer is migrated to a cell.
+		podStore, err := legacy.resources.Pods.Store(legacy.ctx)
+		if err != nil {
+			log.WithError(err).Fatal("Get pod store")
```

Comment:
Do we know why this might return a non nil error? If perhaps we should make the fatal log message more descriptive as to possible reasons for the termination.

Reply:
This is the description of the .Store method:
Store retrieves the read-only store for the resource. Blocks until the store has been synchronized or the context cancelled. Returns a non-nil error if context is cancelled or the resource has been stopped before store has synchronized.

It will only fail if the context is canceled or resource are stopped.
I made the message more descriptive.